flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/CodeGenOperatorFactory.java

```diff
@@ -59,7 +59,12 @@ public ChainingStrategy getChainingStrategy() {
 
 	@Override
 	public boolean isOperatorSelectiveReading() {
-		return InputSelectable.class.isAssignableFrom(generatedClass.getClass(Thread.currentThread().getContextClassLoader()));
+		return InputSelectable.class.isAssignableFrom(getStreamOperatorClass());
+	}
+
+	@Override
+	public Class<? extends StreamOperator> getStreamOperatorClass() {
+		return generatedClass.getClass(Thread.currentThread().getContextClassLoader());
```

Comment:
Is it possible that this code can be triggered in a settings where the context class loader is not a user code class loader?

Reply:
I will add a `Classloader` type parameter to this method.